src/Illuminate/Validation/ValidationRuleParser.php

```diff
@@ -85,8 +85,12 @@ protected function explodeRules($rules)
     protected function explodeExplicitRule($rule, $attribute)
     {
         if (is_string($rule)) {
-            return explode('|', $rule);
-        } elseif (is_object($rule)) {
+            [$name] = static::parseStringRule($rule);
+
+            return explode('|', $rule, ...array_filter([static::ruleIsRegex($name)]));
```

Comment:
I've used the spread operator to completely omit the third (`$limit`) parameter if the rule **is not** regex.

I've added this, because `explode()` will take a `null` or `0` parameter and set the `$limit` to `1`.

We want `$limit` to be `1` when the rule is a regular expression, or `$limit` to be completely omitted when the rule is not.